src/transformers/adapters/mixins/albert.py

```diff
@@ -0,0 +1,44 @@
+from typing import Iterable, Tuple
+
+import torch.nn as nn
+
+from ..layer import AdapterLayer
+from ..model_mixin import (
+    EmbeddingAdaptersMixin,
+    EmbeddingAdaptersWrapperMixin,
+    InvertibleAdaptersMixin,
+    ModelAdaptersMixin,
+    ModelWithHeadsAdaptersMixin,
+)
+
+
+class AlbertAttentionAdaptersMixin:
+    """Adds adapters to the AlbertAttention module of ALBERT."""
+
+    def _init_adapter_modules(self):
+        self.attention_adapters = AdapterLayer("mh_adapter", self.config)
+        self.attention_adapters._init_adapter_modules()
+
+
+class AlbertModelAdaptersMixin(EmbeddingAdaptersMixin, InvertibleAdaptersMixin, ModelAdaptersMixin):
+    """Adds adapters to the AlbertModel module."""
+
+    def iter_layers(self) -> Iterable[Tuple[int, nn.Module]]:
+        for i, albertLayerGroup in enumerate(self.encoder.albert_layer_groups):
+            for albertLayer in albertLayerGroup.albert_layers:
+                yield i, albertLayer
```

Comment:
I'm not sure about the exact logic of Albert's layer sharing, but couldn't treating the layer group id here as layer id for each adapter be problematic for certain config keys (e.g. reduction factor per layer, leave_out)?

Reply:
You are right - should be fixed by acefa59c3b3ebbf65eb92f140ca0fed492e4ed18

Reply:
Thanks for fixing.

I think we should additionally add a note in the docs to clarify how the adapter implementation in Albert is different from others (i.e. that layers + adapters are shared and using layer ids in the adapter config will not affect the actual Albert layers, but the hidden layer groups + inner groups). I did a similar thing for CLIP here: https://raw.githubusercontent.com/adapter-hub/adapter-transformers/master/adapter_docs/classes/models/clip.rst

Reply:
Added a note to the docs in 4d2fb06 :+1: